lib/hathifiles_database/db/connection.rb

```diff
@@ -79,6 +90,14 @@ def delete_existing_data(lines)
         end
       end
 
+      # @param [Enumerable<String>] htids An enumeration of HTIDs.
+      def delete_existing_htids(htids)
```

Comment:
It doesn't seem like the `Set` for deletes buys us much here - we could just read the file of IDs in chunks (`each_slice` as above, perhaps) and do the deletes.